docs/tutorials/streaming_analytics.md

```diff
@@ -0,0 +1,460 @@
+    .keyBy(...)
+    .window(...)
+    .sideOutputLateData(lateTag)
+    .process(...);
+  
+DataStream<Event> lateStream = result.getSideOutput(lateTag);
+{% endhighlight %}
+
+You can also specify an interval of _allowed lateness_ during which the late events will continue to
+be assigned to the appropriate window(s) (whose state will have been retained). By default each late
+event will cause a late firing of the window function.
+
+By default the allowed lateness is 0. In other words, elements behind the watermark are dropped (or
+sent to the side output).
+
+For example:
+
+{% highlight java %}
+stream.
+    .keyBy(...)
+    .window(...)
+    .allowedLateness(Time.seconds(10))
+    .process(...);
+{% endhighlight %}
+
+When the allowed lateness is greater than zero, only those events that are so late that they would
+be dropped are sent to the side output (if it has been configured).
+
+### Surprises
+
+Some aspects of Flink's windowing API may not behave in the way you would expect. Based on
+frequently asked questions on the [flink-user mailing
+list](https://flink.apache.org/community.html#mailing-lists) and elsewhere, here are some facts
+about windows that may surprise you.
+
+#### Sliding Windows Make Copies
+
+Sliding window assigners can create lots of window objects, and will copy each event into every
+relevant window. For example, if you have sliding windows every 15 minutes that are 24-hours in
+length, each event will be copied into 4 * 24 = 96 windows.
+
+#### Time Windows are Aligned to the Epoch
+
+Just because you are using hour-long processing-time windows and start your application running at
+12:05 does not mean that the first window will close at 1:05. The first window will be 55 minutes
+long and close at 1:00.
+
+#### Windows Can Follow Windows
+
+For example, it works to do this:
+
+{% highlight java %}
+stream
+    .keyBy(t -> t.key)
+    .timeWindow(<time specification>)
+    .reduce(<reduce function>)
+    .timeWindowAll(<same time specification>)
+    .reduce(<same reduce function>)
+{% endhighlight %}
+
+You might expect Flink's runtime to be smart enough to do this parallel pre-aggregation for you
+(provided you are using a ReduceFunction or AggregateFunction), but it's not.
+
+#### No Results for Empty TimeWindows
+
+Windows are only created when events are assigned to them. So if there are no events in a given time
+frame, no results will be reported.
+
+#### Late Events Can Cause Late Merges
+
+Session windows are based on an abstraction of windows that can _merge_. Each element is initially
+assigned to a new window, after which windows are merged whenever the gap between them is small
+enough. In this way, a late event can bridge the gap separating two previously separate sessions,
+producing a late merge.
+
+{% top %}
+
+## Hands-on
+
+The hands-on exercise that goes with this section is the [Hourly Tips
+Exercise](https://github.com/apache/flink-training/tree/master/hourly-tips).
```

Comment:
same as in the previous PR: is it possible to link to the right branch?

Reply:
Already done.